src/matrixLabels/gatherTextTriangles.js

```diff
@@ -59,16 +62,33 @@ export default function gather_text_triangles(store, viz_area, inst_axis) {
         if (labels.precalc[inst_axis]) {
           // calculate text vector
           // vectorize the label so we can draw it at any scale
-          inst_text_vect = vectorize_label(store, inst_axis, inst_name);
-          text_triangles[inst_axis][inst_name] = inst_text_vect;
-          // current and new offsets, based on reordering
-          inst_text_vect.inst_offset = [0, inst_label.offsets.inst];
-          inst_text_vect.new_offset = [0, inst_label.offsets.new];
-          text_triangles.draw[inst_axis].push(inst_text_vect);
+          tasks.push(vectorizeWorker.postMessage({
+            fontDetail,
+            name: inst_name,
+            axis: inst_axis,
+            offsetInst: inst_label.offsets.inst,
+            offsetNew: inst_label.offsets.new,
+          }));
+          store.dispatch(
+            dropFromLabelQueue({ queue: "low", axis: inst_axis, label: inst_name })
```

Comment:
Does this maintain functionality? If we originally did this synchronously at the end of vectorizeLabel and now we do it before we receive the results from the async operations, is it the same? Would it be more equivalent to add it into the results.forEach of the .then of the Promise.all a bit later?